src/main/webapp/scripts/place-suggestions.js

```diff
@@ -42,10 +44,110 @@ function findPlacesInCity(city, filter) {
     return;
   }
   
-  console.log(city);
-  console.log(filter);
-} 
+  placeCardsContainer.innerHTML = LOADING_ANIMATION_HTML;
+  const { geometry } = city;
+  const { location } = geometry;
+  const { lat, lng } = location;
+  const placesResponse = await fetch(
+    `https://cors-anywhere.herokuapp.com/https://maps.googleapis.com/maps/api/place/textsearch/json?type=${filter}&location=${lat()},${lng()}&radius=10000&key=${GOOGLE_API_KEY}&output=json`
+  )
+  const { results } = await placesResponse.json();
+  getPlaceCardInformation(results);
+}
 
+async function getPlaceCardInformation(places) {
+  let placeDetailsArr = [];
+  for(let i = 0; i < places.length; i++) {
+    const { place_id } = places[i];
+    const placeDetails = await getPlaceDetails(place_id);
+    placeDetailsArr.push(placeDetails);
+  }
 
+  renderPlaceCards(placeDetailsArr);
+}
 
+function renderPlaceCards(places) {
+  let placeCards = [];
+  for(let i = 0; i < places.length; i++) {   
+    const { phoneNumber, name, photoUrl, priceLevel, rating, address, website } = places[i];
+    const addressClass = address ? '' : 'undefined';
+    const phoneNumberClass = phoneNumber ? '' : 'undefined';
+    const ratingClass = rating ? '' : 'undefined';
+    const priceLevelClass = priceLevel ? '' : 'undefined';
+    const websiteClass = website ? '' : 'undefined';
```

Comment:
Instead of setting the CSS classes to 'undefined' and then removing them, why not just conditionally render them in the template string below?

For example,
```
`
<div class="card-content"> ` +
  addressClass ==="undefined"
  ? ""
  : `<p>${address}</p>` + 

...and so on
```

Reply:
Thank you! I was trying to figure out how to conditionally render HTML elements in JS and this is the best I could I think of at the time lol. I'll def fix this immediately.